### PR TITLE
Fix `length-zero-no-unit` false positives for Sass variables

### PR DESCRIPTION
**Pull request by ryo-manba**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/8034

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/length-zero-no-unit/__tests__/index.mjs [hunk 1]
```diff
@@ -806,38 +812,17 @@ testRule({
...
 		{
 			code: '@detached-ruleset: { font-size: 0px; };',
-			fixed: '@detached-ruleset: { font-size: 0; };',
-
-			message: messages.rejected,
-			line: 1,
-			column: 34,
-			endLine: 1,
-			endColumn: 36,
+			description: 'ignore Less detached ruleset with font-size',
```
Comment (ryo-manba) [inline lib/rules/length-zero-no-unit/__tests__/index.mjs:817]: This change will also prevent variables in `@detached-ruleset` from being updated. 
Do you think this is appropriate?

Comment (jeddy3) [reply lib/rules/length-zero-no-unit/__tests__/index.mjs:817]: Yes, I believe it's fine. These days, we typically prefer to ignore non-standard constructs in the built-in rules.

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you.

**Review by ybiquitous — APPROVED** [review]
@ryo-manba Thanks for the quick fix! LGTM 👍🏼

**Pull request merged**